### ISSUE TITLE
Issues interpreting the final_cltv

### ISSUE THREAD

**Issue by cdecker** [lightningnetwork/lightning-rfc#268]
Hunting down some of the issues from integration test run cdecker/lightning-integration#7. It seems we are interpreting the `final_cltv_expiry` passed in the payment request in a number of different, and incompatible ways:

c-lightning interprets anything that is smaller or equal to the current blockheight + `final_cltv` as invalid and rejects it. `lnd` interprets the `final_cltv` as the "final delta should be exactly that", which makes it equal to the drop-to-chain threshold for c-lightning. When sending a payment, c-lightning adds some slack (in the form of a +1 to the delta). Finally eclair uses a fixed value of 9 (with the tested version in cdecker/lightning-integration#7), which is bigger than c-lightning's default value of 6+slack and thus succeeds.

Is the final_cltv to be interpreted as "greater than or equal" or as an absolute "greater than"?

**Comment by pm47** [comment]
The value specified in the payment request is a *minimum* (name of the field is actually `min_final_cltv_expiry`).

**Comment by cdecker** [comment]
So the interpretation should be "anything lower than this will be considered illegal, but being equal is fine", seems reasonable.

**Comment by Roasbeef** [comment]
Yeah the reason `lnd` uses the _exact_ value is that we don't yet add any additional shadow route to the end of a crafted HTLC route. When _accepting_ HTLC's as the final hop, we interpret is as a _minimum_.
